waspc/src/Wasp/Generator/DbGenerator/Common.hs

```diff
@@ -60,3 +66,42 @@ dbSchemaChecksumOnLastGenerateFileInDbRootDir = [relfile|schema.prisma.wasp-gene
 
 dbSchemaChecksumOnLastGenerateFileProjectRootDir :: Path' (Rel ProjectRootDir) (File DbSchemaChecksumOnLastGenerateFile)
 dbSchemaChecksumOnLastGenerateFileProjectRootDir = dbRootDirInProjectRootDir </> dbSchemaChecksumOnLastGenerateFileInDbRootDir
+
+data MigrateArgs = MigrateArgs
+  { _migrationName :: Maybe String,
+    _isCreateOnlyMigration :: Bool
+  }
+  deriving (Show, Eq)
+
+emptyMigrateArgs :: MigrateArgs
+emptyMigrateArgs = MigrateArgs {_migrationName = Nothing, _isCreateOnlyMigration = False}
+
+asArgs :: MigrateArgs -> [String]
+asArgs migrateArgs = do
+  concat . concat $
+    [ [["--create-only"] | _isCreateOnlyMigration migrateArgs],
+      [["--name", name] | Just name <- [_migrationName migrateArgs]]
+    ]
+
+parseMigrateArgs :: Maybe [String] -> MigrateArgs
+parseMigrateArgs Nothing = emptyMigrateArgs
+parseMigrateArgs (Just migrateArgs) = do
+  go migrateArgs emptyMigrateArgs
+  where
+    go :: [String] -> MigrateArgs -> MigrateArgs
+    go ("--create-only" : rest) mArgs = go rest mArgs {_isCreateOnlyMigration = True}
```

Comment:
Niiiceeee, proper fun recursion :D! Yeah this what we are about right :D!?

Reply:
I know, right! Haskell, baby!!! It has sadly been so long since I wrote a fun little function like this that it took me a second to get my mind into it though lol 😅